notebooks/Notebook2.md

```diff
@@ -2,33 +2,56 @@
+
+I began the week by building a physical java interface for the system that would allow the user to input the language into a text box. This allowed me to easily test inputs and see if it was working properly. This went fairly smoothly until I tried to add a part to the language, this was a simple “start new game log” phrase that was evaluated in the interface portion of the code and created a new game. This would then allow the user to input the regular dsl language. However I kept getting a null pointer error and I could not figure out why because there were no null pointers being passed through. Eventually I figured out that when adding this extra piece of code to create a new game I was building a null game. I ended taking that portion out and it worked great from there on. I may end up going back and retrying something similar later. ([code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/blob/main/src/Interface/TextInputInterface.java))
+
+Next I started to simplify the player type classes with individual setters per input type vs one big setter that sets all input types. This was fairly straightforward, I got rid of alot of bulk in the abstract player class and allowed for each player type to be created with just the input of the players number instead of name and number and all the other components that each player has. Then I built setters for each player's unique inputs. This allowed me to write something similar to the following:
+New PassingPlayer(playerOrNum).setYards(numYards).setINCs(outcome).setTDs(outcome).setINTs(outcome).updatePlayer();
+This was useful because I was able to handle inputs that are the same across different methods. ([code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/tree/main/src/Player))
+
+Finally I built a parser that takes each part of the input language and utilizes the Game class add function to add a player into the data structure. This involved building code to determine what type of playtype each input from the user was. These functions are my is…() functions. They return a boolean value if the input matches a possible input string. This could be further simplified by using another type of parser but for now It works great. Next I worked on function that are designed to add the player to the game and return the game to the parser these function utilize the add function from the game class to add a new player to the data structure or if the player already exists update the stats by merging the players stats that were just created with the old ones. The individualized settlers for each player type came in really nice for this aspect. Finally I worked on some functions that would transform strings to numbers and parsing out yardage from a string. For example, if the user inputs 22y representing 22 yards the parser needs to transform that into just an integer number 22. ([code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/blob/main/src/Parser/Parser.java))
```

Comment:
Great work on the parser, it seems like you were able to capture a lot of the cases you were planning in your original design. I looked over it and it does seem like most of the parsing code was good I liked the way you utilized groups to split each part of the input. One note I would have is that your StringGrouping.java file contains `if(groupsMatcher.matches())` which works to prevent errors but it seems like nothing happens if a user inputs a string that doesn't match the regex pattern so I would recommend adding an else case which lets users know they've made an error